src/Data/Vector/Persistent.hs

```diff
@@ -236,10 +240,17 @@ pvTraverse f = go
   where
     go (RootNode sz sh t as)
       | sz == 0 = Ap.pure empty
-      | otherwise = Ap.liftA2 (RootNode sz sh) (T.traverse f t) (A.traverseArray go_ as)
+      | otherwise = Ap.liftA2 (\as' t' -> RootNode sz sh t' as') (A.traverseArray go_ as) (forwards $ T.traverse (liftBackwards f) t)
     go_ (DataNode a) = DataNode Ap.<$> A.traverseArray f a
     go_ (InternalNode as) = InternalNode Ap.<$> A.traverseArray go_ as
 
+    liftBackwards :: (a -> g b) -> a -> Backwards g b
+#if __GLASGOW_HASKELL__ >= 708
+    liftBackwards = coerce
```

Comment:
Does this actually win us anything for `Backwards`? I put it in on instinct, but we should check the Core.